File: lab2/Model1.vb
Imports System
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.Data.Entity
Imports System.Linq

Partial Public Class Model1
	Inherits DbContext

	Public Sub New()
		MyBase.New("name=Model1")
	End Sub

	Public Overridable Property Evaluations As DbSet(Of Evaluations)
	Public Overridable Property Specialties As DbSet(Of Specialties)
	Public Overridable Property Students As DbSet(Of Students)
	Public Overridable Property Subjects As DbSet(Of Subjects)
	Public Overridable Property sysdiagrams As DbSet(Of sysdiagrams)

	Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
		modelBuilder.Entity(Of Specialties)() _
			.HasMany(Function(e) e.Students) _
			.WithOptional(Function(e) e.Specialties) _
			.WillCascadeOnDelete()

		modelBuilder.Entity(Of Students)() _
			.HasOptional(Function(e) e.Evaluations) _
			.WithRequired(Function(e) e.Students) _
			.WillCascadeOnDelete()

		modelBuilder.Entity(Of Subjects)() _
			.HasMany(Function(e) e.Evaluations) _
			.WithOptional(Function(e) e.Subjects) _
			.HasForeignKey(Function(e) e.ID_Subjects1)

		modelBuilder.Entity(Of Subjects)() _
			.HasMany(Function(e) e.Evaluations1) _
			.WithOptional(Function(e) e.Subjects1) _
			.HasForeignKey(Function(e) e.ID_Subjects2) _
			.WillCascadeOnDelete()
	End Sub
End Class

File: Labs2/Models/Model1.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using System.Data.Entity;
using System.Linq;

namespace Labs2.Models
{
    public partial class Model1 : DbContext
    {
        public Model1()
            : base("name=Model1")
        {
        }

        public virtual DbSet<Evaluations> Evaluations { get; set; }
        public virtual DbSet<Specialties> Specialties { get; set; }
        public virtual DbSet<Students> Students { get; set; }
        public virtual DbSet<Subjects> Subjects { get; set; }
        public virtual DbSet<sysdiagrams> sysdiagrams { get; set; }

        protected override void OnModelCreating(DbModelBuilder modelBuilder)
        {
            modelBuilder.Entity<Specialties>()
                .HasMany(e => e.Students)
                .WithOptional(e => e.Specialties)
                .WillCascadeOnDelete();

            modelBuilder.Entity<Students>()
                .HasOptional(e => e.Evaluations)
                .WithRequired(e => e.Students)
                .WillCascadeOnDelete();

            modelBuilder.Entity<Subjects>()
                .HasMany(e => e.Evaluations)
                .WithOptional(e => e.Subjects)
                .HasForeignKey(e => e.ID_Subjects1);

            modelBuilder.Entity<Subjects>()
                .HasMany(e => e.Evaluations1)
                .WithOptional(e => e.Subjects1)
                .HasForeignKey(e => e.ID_Subjects2)
                .WillCascadeOnDelete();
        }
    }
}
